app/src/server/trpc/routers/service.ts

```diff
@@ -16,5 +20,33 @@ export const serviceRouter = router({
       Measure: measureBundle.total ?? 0,
       Library: libraryBundle.total ?? 0
     } as const;
-  })
+  }),
+
+  getArtifactsByResource: publicProcedure
+    .input(z.object({ resourceType: z.enum(['Measure', 'Library']) }))
+    .query(async ({ input }) => {
+      const artifactBundle = await fetch(`${process.env.NEXT_PUBLIC_MRS_SERVER}/${input.resourceType}`).then(
+        resArtifacts => resArtifacts.json() as Promise<fhir4.Bundle<FhirArtifact>>
+      );
+
+      const artifactList = artifactBundle.entry?.map(entry => ({
+        label: entry.resource?.name || entry.resource?.id || '',
+        value: entry.resource?.id || `${entry.resource?.resourceType}` || ''
+      }));
+
+      return artifactList;
+    }),
+
+  convertArtifactById: publicProcedure
+    .input(z.object({ resourceType: z.enum(['Measure', 'Library']), id: z.string() }))
+    .mutation(async ({ input }) => {
+      const draftRes = await fetch(`${process.env.NEXT_PUBLIC_MRS_SERVER}/${input.resourceType}/${input.id}`);
+      const draftArtifact = (await draftRes.json()) as FhirArtifact;
+      draftArtifact.id = uuidv4();
+      draftArtifact.status = 'draft';
+
+      const draftCaller = draftRouter.createCaller({});
+      const res = await draftCaller.createDraft({ resourceType: input.resourceType, draft: draftArtifact });
```

Comment:
Still trying to wrap my head around `createCaller` since this is my first time seeing it used in context. I was reading through the [tRPC docs](https://trpc.io/docs/server/server-side-calls) on server-side calls, which says the following: 

> createCaller should not be used to call procedures from within other procedures. This creates overhead by (potentially) creating context again, executing all middlewares, and validating the input - all of which were already done by the current procedure.

Are we doing this here or not? Seems like `draftRouter` has a procedure `createDraft`, and then `serviceRouter` now has this procedure `getArtifactsByResource` that calls the `createDraft` procedure.

Reply:
Ahh for some reason I thought this was not doing that but I think you are right! Not quite sure how to work around this, I will look into it